### ISSUE TITLE
Issue with compiling with multiple inline partials.

### ISSUE THREAD

**Issue by cjbarth** [zordius/lightncandy#301]
The PHP Code:
-------------

```php
require('./vendor/autoload.php');
use LightnCandy\LightnCandy;

// The Template:
$template = <<<VAREND
{{#*inline "t1"}}
    {{#if imageUrl}}
        <span />
    {{else}}
        <div />
    {{/if}}
{{/inline}}
{{#*inline "t2"}}
    {{#if imageUrl}}
        <span />
    {{else}}
        <div />
    {{/if}}
{{/inline}}
{{#*inline "t3"}}
    {{#if imageUrl}}
        <span />
    {{else}}
        <div />
    {{/if}}
{{/inline}}


<div>
</div>

VAREND;

// Helpers:
$helpers = array(
  'foo' => function () {
    return 'Hello World';
  },
);

// Partials:
$partials = array(
    'bar' => 'Yes, partial',
);

$phpStr = LightnCandy::compile($template, array(
  // Used compile flags
  'flags' => LightnCandy::FLAG_ERROR_EXCEPTION | LightnCandy::FLAG_HANDLEBARS | LightnCandy::FLAG_RUNTIMEPARTIAL,
  'helpers' => $helpers,
  'partials' => $partials,
));

echo "Generated PHP Code:\n$phpStr\n";

// Input Data:
$data = array(
  'qoo' => 'moo'
);

// Save the compiled PHP code into a php file
file_put_contents('render.php', '<?php ' . $phpStr . '?>');

// Get the render function from the php file
$renderer = include('render.php');

echo "Result:\n" . $renderer($data);
```

The Issue:
----------

I get the following compilation error:

```Fatal error: Uncaught Exception: {{else}} only valid in if, unless, each, and #section context in /Users/.../LightnCandy.php:110
Stack trace:
#0 /Users/.../LightnCandy.php(54): LightnCandy\LightnCandy::handleError(Array)
#1 /Users/.../compile-handlebars-templates.php(55): LightnCandy\LightnCandy::compile('\t\n{{#*inline "t...', Array)
#2 {main}
  thrown in /Users/.../LightnCandy.php on line 110
```

However, if I remove any one of the definitions for partials in the template, `t1` or `t2` or `t3`, which are all identical, complication and rendering work just fine.

**Comment by cjbarth** [comment]
It appears that this problem can be fixed with the following change:

```
    protected static function blockEnd(&$context, &$vars, $match = null) {
// we shouldn't remove a level if we didn't match what we wanted
//        $context['level']--;
        $c = count($context['stack']) - 2;
        $pop = ($c >= 0) ? $context['stack'][$c + 1] : '';
        if (($match !== null) && ($match !== $pop)) {
            return;
        }
        $pop2 = ($c >= 0) ? $context['stack'][$c]: '';
        switch ($context['currentToken'][Token::POS_INNERTAG]) {
            case 'with':
                if (!$context['flags']['nohbh']) {
                    if ($pop2 !== '[with]') {
                        $context['error'][] = 'Unexpect token: {{/with}} !';
                        return;
                    }
                }
// we matched what we wanted, remove a level
	            $context['level']--;
                return true;
        }

        switch($pop) {
            case '#':
            case '^':
                $elsechain = array_shift($context['elselvl']);
                if (isset($elsechain[0])) {
// we are in an else chain, but haven't remove any levels, so don't add one back in
//                    $context['level']++;
                    $context['currentToken'][Token::POS_RSPACE] = $context['currentToken'][Token::POS_BACKFILL] = '{{/' . implode('}}{{/', $elsechain) . '}}' . Token::toString($context['currentToken']) . $context['currentToken'][Token::POS_RSPACE];
                    return Token::POS_BACKFILL;
                }
            case '#>':
            case '#*':
                list($levels, $spvar, $var) = Expression::analyze($context, $vars[0]);
                $v = Expression::toString($levels, $spvar, $var);
                if ($pop2 !== $v) {
                    $context['error'][] = 'Unexpect token ' . Token::toString($context['currentToken']) . " ! Previous token {{{$pop}$pop2}} is not closed";
                    return;
                }
// we found what we wanted, remove a level
		        $context['level']--;
                return true;
            default:
                $context['error'][] = 'Unexpect token: ' . Token::toString($context['currentToken']) . ' !';
                return;
        }
    }
```

I'll be glad to create a PR for this if desired.

**Comment by zordius** [comment]
Thank you for your reporting! Will check on this issue later.

**Comment by zordius** [comment]
This is fixed by #302, thank you very much!!
